scripts/yast2

```diff
@@ -290,10 +290,11 @@ fi
 if [ -n "$CMDLINE_HELP" ]; then
     set -- $@ help;
 fi
-if [ $# -gt 0 ]; then
-    a=("$@")
-    set -- --arg "${a[@]}"
-fi
+ARGS=""
+for i in "$@"; do
+  ARGS="$ARGS --arg '$i'"
```

Comment:
Bash is awful, but it does have a solution for this: `printf %q` is the proper way to quote for an `eval`:

```sh
ARGS=$(printf '%s -arg %q' "$ARGS" "$i")
```